sway-lib-std/src/chain/auth.sw

```diff
@@ -61,28 +62,33 @@ fn get_coins_owner() -> Result<Identity, AuthError> {
             i += 1;
         } else {
             // type == InputCoin
-            let input_owner = Option::Some(tx_input_coin_owner(i));
+            let input_owner = tx_input_owner(i);
             if candidate.is_none() {
                 // This is the first input seen of the correct type.
                 candidate = input_owner;
                 i += 1;
             } else {
                 // Compare current coin owner to candidate.
-                // `candidate` and `input_owner` must be `Option::Some` at this point,
-                // so can unwrap safely.
+                // `candidate` and `input_owner` must be `Option::Some`
+                // at this point, so we can unwrap safely.
                 if input_owner.unwrap() == candidate.unwrap() {
                     // Owners are a match, continue looping.
                     i += 1;
                 } else {
                     // Owners don't match. Return Err.
+                    // TODO: Use break keyword when possible
                     i = inputs_count;
                     return Result::Err(AuthError::InputsNotAllOwnedBySameAddress);
                 };
             };
-        };
+        }
     }
 
+    // we want to return an error here rather than reverting
+    if candidate.is_none() {
+        return Result::Err(AuthError::NoCoinOrMessageInputs);
+    };
+
```

Comment:
I might be able to remove this, as _I think_ the only way `candidate` can be `None` here is on an invalid tx (which shouldn't happen. (i.e: a tx with only a single input which is **not** an `InputCoin`).

Reply:
A transaction must have at least one inputCoin for fees, so I agree that you can safely remove this.

Reply:
+1. If you remove this though then there's no need for `AuthError::NoCoinOrMessageInputs`, right?

Reply:
correct, I'll be removing that error variant.